extensions/amp-inline-gallery/amp-inline-gallery.md

```diff
@@ -0,0 +1,225 @@
+
+Pagination can be specified as either inset (overlaying the carousel) or outset, underneath the carousel. If you want to use different styles for different screen sizes, you can do something like the following:
+
+```html
+<amp-inline-gallery layout="container">
+  <amp-base-carousel>…</amp-base-carousel>
+  <amp-inline-gallery-pagination
+    media="(max-width: 599px)"
+    layout="nodisplay"
+    inset
+  >
+  </amp-inline-gallery-pagination>
+  <amp-inline-gallery-pagination
+    media="(min-width: 600px)"
+    layout="fixed-height"
+    height="24"
+  >
+  </amp-inline-gallery-pagination>
+</amp-inline-gallery>
+```
+
+Where `media` is a [CSS Media Query](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries), which controls which pagination configuration to display.
+
+### Using thumbnails
+
+The inline gallery can display thumbnail previews in addition or instead of the pagination indicators. When using thumbnails, some things to keep in mind are:
+
+- You will likely want to avoid using both pagination indicators and thumbnails, unless you have eight or more thumbnails, as the indicator dots may be redundant with the thumbnails for determining progress for smaller galleries.
+- When showing both inset pagination indicators and thumbnails, you will likely want to have the pagination overlap the slide as shown below.
+- You can use the `media` attribute to show pagination on smaller mobile devices and thumbnails on larger screens.
+- You will want to likely want to use `srcset`, allowing the thumbnails to use a lower resolution image that can load much more quickly. You can omit the `sizes` on `<amp-img>`, one will be generated automatically depending on the rendered width.
+
+Keeping the above in mind, usage may look like:
+
+```html
+<amp-inline-gallery layout="container">
+  <!-- Used to have the pagination display on top of the images. -->
+  <amp-layout layout="container">
+    <amp-base-carousel
+      class="gallery"
+      layout="responsive"
+      width="3"
+      height="2"
+      snap-align="center"
+      loop="true"
+    >
+      <amp-img
+        class="slide"
+        layout="flex-item"
+        src="<large-img-url>"
+        srcset="<thumbnail-image-url> 150w,
+                    <medium-image-url> 600w,
+                    <large-image-url> 1200w"
+      >
+      </amp-img>
+      <!-- More slides -->
+    </amp-base-carousel>
+    <!--
+        If using fewer than 8 slides, consider adding something
+        like media="(max-width: 799px)".
+      -->
+    <amp-inline-gallery-pagination layout="nodisplay" inset>
+    </amp-inline-gallery-pagination>
+  </amp-layout>
+  <amp-inline-gallery-thumbnails
+    media="(min-width: 800px)"
+    layout="fixed-height"
+    height="96"
+  >
+  </amp-inline-gallery-thumbnails>
+</amp-inline-gallery>
+```
+
```

Comment:
Insert section:
`#### amp-inline-gallery-thumbnails attributes`

`##### aspect-ratio-height (optional)`

Specifies the aspect ratio of thumbnails when used with `aspect-ratio-width`. The aspect radio defaults to match the slides in `<amp-base-carousel>`. 

`#### aspect-ratio-width (optional)`
Specified the aspect ratio when used with `aspect-ratio-height`. The aspect radio defaults to match the slides in `<amp-base-carousel>`. 

`#### loop (optional)`
Loops thumbnails. Takes a value of `"true"` or `"false"`. Defaults to `"true"`.

##### Common attributes
The `<amp-inline-gallery-pagination>` element includes [common attributes](https://amp.dev/documentation/guides-and-tutorials/learn/common_attributes) extended to AMP components.